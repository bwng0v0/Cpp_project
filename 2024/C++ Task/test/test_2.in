5
9
1
1
0
2
3
4
5
6
0
0
0
