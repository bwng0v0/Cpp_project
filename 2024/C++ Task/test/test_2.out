
******************************* Main Menu *********************************
* 0.Exit 1.PersonManager(ch3_2, 4, 6, 7_2, 8, 9, 10)                      *
* 2.Class:Object(ch3_1) 3.CopyConstructor(ch5_1) 4.AllocatedMember(ch5_2) *
* 5.OperatorOverload(ch7_1) 6.Inheritance(ch8) 7.PMbyVector               *
***************************************************************************
Menu item number? 5

+++++++++++++++++++ Operator Overload +++++++++++++++++
+ 0.Exit 1.MemoAdd 2.Equal 3.Add 4.Assign 5.Increment +
+ 6.Shift 7.TypeConversion 8.CurrentUser 9.VectorOOL  +
+++++++++++++++++++++++++++++++++++++++++++++++++++++++
Menu item number? 9
pv1: count 2
[0] Hong  0 72.1 true ::
[1] Mong  1 65.4 true ::

pv2: count 3
[0] Choon 3 56.7 true ::
[1] Soon  2 87.6 true ::
[2] Chung 4 67.8 true ::


+++++++++++++ Vector Operator Overload ++++++++++++++
+ 0.Exit 1.operator[] 2.operator! 3.CopyConstructor +
+ 4.operator= 5.operator+ 6.operator=+              +
+++++++++++++++++++++++++++++++++++++++++++++++++++++
Menu item number? 1

====================== Person Management Menu ===================
= 0.Exit 1.Display 2.Append 3.Clear 4.Login(CurrentUser, ch9)   =
= 5.Insert(6_2) 6.Delete(6_2) 7.CopyPersons(7_3) 8.Reset(7_3)   =
= 9.Find(9_2) 10.DispAlbaStud(9_2) 11.DispPhones(9_2)           =
=================================================================
Menu item number? 1
display(): count 16
[0] p0 10 70 false :Gwangju Nam-gu Bongseon-dong 21:
[1] p1 11 61.1 true :Jong-ro 1-gil, Jongno-gu, Seoul:
[2] p2 12 52.2 false :1001, Jungang-daero, Yeonje-gu, Busan:
[3] p3 13 83.3 true :100 Dunsan-ro Seo-gu Daejeon:
[4] s1 21 65.4 true :Jongno-gu Seoul: Physics 3.8 1
[5] s2 22 54.3 false :Yeonje-gu Busan: Electronics 2.5 4
[6] s3 23 55.5 true :Dong-gu Incheon: Computer 3.5 2
[7] s4 24 66.6 false :Sasang-gu Sejong: History 3.1 1
[8] w1 31 33.3 false :Kangnam-gu Seoul: Samsung Director
[9] w2 32 44.4 true :Dobong-gu Kwangju: Hyundai Manager
[10] w3 33 55.5 true :Dong-gu Incheon: Hyundai Labor
[11] w4 34 66.6 false :Sasang-gu Sejong: Kia CEO
[12] a1 41 55.5 true :Dong-gu Incheon: Computer 3.5 2 Hyundai Labor :CU KangNam,Seven Eleven,GSStore Suwon: false
[13] a2 42 66.6 false :Sasang-gu Sejong: History 3.1 1 Kia CEO :Seven Eveven,eMart Jinju,CU Bongsun: true
[14] a3 43 51.9 false :Buk-gu Daejeon: Computer 3.5 2 Samsung Manager :Youngpung,Kyobo Gwanghwa,E-mart Suwon: true
[15] a4 44 66.6 true :Nam-gu Busan: History 3.1 1 LG DepartmentHead :CU, FamilyMart, LotteMart, HomePlus: false

====================== Person Management Menu ===================
= 0.Exit 1.Display 2.Append 3.Clear 4.Login(CurrentUser, ch9)   =
= 5.Insert(6_2) 6.Delete(6_2) 7.CopyPersons(7_3) 8.Reset(7_3)   =
= 9.Find(9_2) 10.DispAlbaStud(9_2) 11.DispPhones(9_2)           =
=================================================================
Menu item number? 0

+++++++++++++ Vector Operator Overload ++++++++++++++
+ 0.Exit 1.operator[] 2.operator! 3.CopyConstructor +
+ 4.operator= 5.operator+ 6.operator=+              +
+++++++++++++++++++++++++++++++++++++++++++++++++++++
Menu item number? 2
count 0

if(pv): false
operator bool(): false
if(!pv): true
operator !(): true
count 1
[0] Hong  0 72.1 true ::

(pv? true: false) true
count 2
[0] Hong  0 72.1 true ::
[1] Chung 2 67.8 true ::

pv[0]: Hong  0 72.1 true ::
delete pv[1];

+++++++++++++ Vector Operator Overload ++++++++++++++
+ 0.Exit 1.operator[] 2.operator! 3.CopyConstructor +
+ 4.operator= 5.operator+ 6.operator=+              +
+++++++++++++++++++++++++++++++++++++++++++++++++++++
Menu item number? 3
Vector<Person*> pv3 = pv2
pv3: count 3
[0] Choon 3 56.7 true ::
[1] Soon  2 87.6 true ::
[2] Chung 4 67.8 true ::

pv3.erase(0)
pv3: count 2
[0] Soon  2 87.6 true ::
[1] Chung 4 67.8 true ::

pv2: count 3
[0] Choon 3 56.7 true ::
[1] Soon  2 87.6 true ::
[2] Chung 4 67.8 true ::

disp_vector(call_return_value(vp2))
pv: count 3
[0] Choon 3 56.7 true ::
[1] Soon  2 87.6 true ::
[2] Chung 4 67.8 true ::

return pv1 
count 2
[0] Hong  0 72.1 true ::
[1] Mong  1 65.4 true ::

pv1: count 2
[0] Hong  0 72.1 true ::
[1] Mong  1 65.4 true ::


+++++++++++++ Vector Operator Overload ++++++++++++++
+ 0.Exit 1.operator[] 2.operator! 3.CopyConstructor +
+ 4.operator= 5.operator+ 6.operator=+              +
+++++++++++++++++++++++++++++++++++++++++++++++++++++
Menu item number? 4
Vector<Person*> pv3 = pv2
pv3: count 3
[0] Choon 3 56.7 true ::
[1] Soon  2 87.6 true ::
[2] Chung 4 67.8 true ::

pv3 = pv1
pv3: count 2
[0] Hong  0 72.1 true ::
[1] Mong  1 65.4 true ::

repeat 9 times: pv3.push_back(pa+2)
pv3: count 11
[0] Hong  0 72.1 true ::
[1] Mong  1 65.4 true ::
[2] Choon 3 56.7 true ::
[3] Choon 3 56.7 true ::
[4] Choon 3 56.7 true ::
[5] Choon 3 56.7 true ::
[6] Choon 3 56.7 true ::
[7] Choon 3 56.7 true ::
[8] Choon 3 56.7 true ::
[9] Choon 3 56.7 true ::
[10] Choon 3 56.7 true ::

pv4 = pv3
pv4: count 11
[0] Hong  0 72.1 true ::
[1] Mong  1 65.4 true ::
[2] Choon 3 56.7 true ::
[3] Choon 3 56.7 true ::
[4] Choon 3 56.7 true ::
[5] Choon 3 56.7 true ::
[6] Choon 3 56.7 true ::
[7] Choon 3 56.7 true ::
[8] Choon 3 56.7 true ::
[9] Choon 3 56.7 true ::
[10] Choon 3 56.7 true ::


+++++++++++++ Vector Operator Overload ++++++++++++++
+ 0.Exit 1.operator[] 2.operator! 3.CopyConstructor +
+ 4.operator= 5.operator+ 6.operator=+              +
+++++++++++++++++++++++++++++++++++++++++++++++++++++
Menu item number? 5
pv3 = pv1 + pv2
pv3: count 5
[0] Hong  0 72.1 true ::
[1] Mong  1 65.4 true ::
[2] Choon 3 56.7 true ::
[3] Soon  2 87.6 true ::
[4] Chung 4 67.8 true ::

pv3 = pv2 + pv2 + pv3
pv3: count 11
[0] Choon 3 56.7 true ::
[1] Soon  2 87.6 true ::
[2] Chung 4 67.8 true ::
[3] Choon 3 56.7 true ::
[4] Soon  2 87.6 true ::
[5] Chung 4 67.8 true ::
[6] Hong  0 72.1 true ::
[7] Mong  1 65.4 true ::
[8] Choon 3 56.7 true ::
[9] Soon  2 87.6 true ::
[10] Chung 4 67.8 true ::


+++++++++++++ Vector Operator Overload ++++++++++++++
+ 0.Exit 1.operator[] 2.operator! 3.CopyConstructor +
+ 4.operator= 5.operator+ 6.operator=+              +
+++++++++++++++++++++++++++++++++++++++++++++++++++++
Menu item number? 6
Vector<Person*> pv4 = pv1: 
pv4: count 2
[0] Hong  0 72.1 true ::
[1] Mong  1 65.4 true ::

pv4 += pv2
pv4: count 5
[0] Hong  0 72.1 true ::
[1] Mong  1 65.4 true ::
[2] Choon 3 56.7 true ::
[3] Soon  2 87.6 true ::
[4] Chung 4 67.8 true ::

pv4 += pv2 + pv2
pv4: count 11
[0] Hong  0 72.1 true ::
[1] Mong  1 65.4 true ::
[2] Choon 3 56.7 true ::
[3] Soon  2 87.6 true ::
[4] Chung 4 67.8 true ::
[5] Choon 3 56.7 true ::
[6] Soon  2 87.6 true ::
[7] Chung 4 67.8 true ::
[8] Choon 3 56.7 true ::
[9] Soon  2 87.6 true ::
[10] Chung 4 67.8 true ::


+++++++++++++ Vector Operator Overload ++++++++++++++
+ 0.Exit 1.operator[] 2.operator! 3.CopyConstructor +
+ 4.operator= 5.operator+ 6.operator=+              +
+++++++++++++++++++++++++++++++++++++++++++++++++++++
Menu item number? 0

+++++++++++++++++++ Operator Overload +++++++++++++++++
+ 0.Exit 1.MemoAdd 2.Equal 3.Add 4.Assign 5.Increment +
+ 6.Shift 7.TypeConversion 8.CurrentUser 9.VectorOOL  +
+++++++++++++++++++++++++++++++++++++++++++++++++++++++
Menu item number? 0

******************************* Main Menu *********************************
* 0.Exit 1.PersonManager(ch3_2, 4, 6, 7_2, 8, 9, 10)                      *
* 2.Class:Object(ch3_1) 3.CopyConstructor(ch5_1) 4.AllocatedMember(ch5_2) *
* 5.OperatorOverload(ch7_1) 6.Inheritance(ch8) 7.PMbyVector               *
***************************************************************************
Menu item number? 0
Good bye!!
